code/modules/religion/rites.dm

```diff
@@ -102,3 +102,116 @@
 	human2borg.set_species(/datum/species/android)
 	human2borg.visible_message("<span class='notice'>[human2borg] has been converted by the rite of [name]!</span>")
 	return TRUE
+
+
+/*********Ever-Burning Candle**********/
+
+///apply a bunch of fire immunity effect to clothing
+/datum/religion_rites/fireproof/proc/apply_fireproof(obj/item/clothing/fireproofed)
+	fireproofed.name = "unmelting [fireproofed.name]"
+	fireproofed.max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT
+	fireproofed.heat_protection = chosen_clothing.body_parts_covered
+	fireproofed.resistance_flags |= FIRE_PROOF
+
+/datum/religion_rites/fireproof
+	name = "Unmelting Wax"
+	desc = "Grants fire immunity to any piece of clothing."
+	ritual_length = 15 SECONDS
+	ritual_invocations = list("And so to support the holder of the Ever-Burning candle...",
+	"... allow this unworthy apparel to serve you ...",
+	"... make it strong enough to burn a thousand time and more ...")
+	invoke_msg = "... Come forth in your new form, and join the unmelting wax of the one true flame!"
+	favor_cost = 1000
+///the piece of clothing that will be fireproofed, only one per rite
+	var/obj/item/clothing/chosen_clothing
+
+/datum/religion_rites/fireproof/perform_rite(mob/living/user, atom/religious_tool)
+	for(var/obj/item/clothing/apparel in get_turf(religious_tool))
+		if(apparel.max_heat_protection_temperature >= FIRE_IMMUNITY_MAX_TEMP_PROTECT)
+			continue //we ignore anything that is already fireproof
+		chosen_clothing = apparel //the apparel has been chosen by our lord and savior
+		return ..()
+	return FALSE
+
+/datum/religion_rites/fireproof/invoke_effect(mob/living/user, atom/religious_tool)
+	for(chosen_clothing in get_turf(religious_tool))
```

Comment:
can you explain what you are trying to do here? After rereading this it doesnt make sense.

Chosen clothing by this point should be a singular item you have saved during perform rite, you shouldnt have to loop through anything.

you might be looking for something like `(!QDELETING(chosen_clothing) && get_turf(religious_tool) == chosen_clothing.loc) //we use loc on the second check since we want to make sure its out in the open on the tool aka altar still`